src/libs/actions/Policy.ts

```diff
@@ -456,7 +457,9 @@ function setWorkspaceAutoReporting(policyID: string, enabled: boolean, frequency
             onyxMethod: Onyx.METHOD.MERGE,
             key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
             value: {
-                autoReporting: !enabled,
+                autoReporting: policy.autoReporting,
+                harvesting: policy.harvesting,
+                autoReportingFrequency: policy.autoReportingFrequency,
```

Comment:
```suggestion
                autoReporting: policy.autoReporting ?? null,
                harvesting: {enabled: policy.harvesting?.enabled ?? null},
                autoReportingFrequency: policy.autoReportingFrequency ?? null,
```

- When merging we need to be sure we are not merging a `undefined` value because it's a NOOP. Merging `{autoReporting: undefined}` is same as not merging at all and the value won't get reverted.
- For objects we should merge only the values that we did actually change and not the whole object

(Same goes for other fields)

Reply:
Got it. Updating